includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -2523,7 +2523,9 @@ public function get_balance_transaction_id_from_charge( $charge ) {
 	 * @return string The order's unique signature. Format: order_id:md5(order_id-order_key-customer_id-order_total).
 	 */
 	protected function get_order_signature( $order ) {
-		$order = ! is_a( $order, 'WC_Order' ) ? wc_get_order( $order ) : $order;
+		if ( is_a( $order, 'WC_Order_Refund' ) ) {
+			$order = wc_get_order( $order->get_parent_id() );
+		}
```

Comment:
The current implementation allows `$order` to also be an `order_id` via `wc_get_order( $order )`... I think we should keep that behavior (or add error handling):
```suggestion
		if ( is_a( $order, 'WC_Order_Refund' ) ) {
			$order = wc_get_order( $order->get_parent_id() );
		} elseif ( ! is_a( $order, 'WC_Order' ) ) {
			$order = wc_get_order( $order );
		}
```